### PR TITLE
Add some comments for the compact proposal feature.

### PR DESCRIPTION
**Pull request by chaoticlonghair**

### CODE REVIEW

**Review by kaikai1024 — COMMENTED** [review]
transaction or tx, choose one

File: src/core/cita_bft.rs [hunk 1]
```diff
@@ -146,6 +146,7 @@ pub struct Bft {
     block_txs: VecDeque<(usize, BlockTxs)>,
     block_proof: Option<(usize, BlockWithProof)>,
 
+    // The verified blocks with tx body.
     verified_blocks: HashMap<H256, Block>,
```
Comment (u2) [inline src/core/cita_bft.rs:150]: So this only can be inserted when receiving the Auth's verified result?

File: src/core/cita_bft.rs [hunk 2]
```diff
@@ -1401,6 +1407,8 @@ impl Bft {
                 });
             }
         } else if tminfo.step == Step::PrecommitAuth {
+            // If consensus doesn't receive the result of block verification in a specific
+            // time-frame，use the original message to construct a request, then resend it to auth.
```
Comment (u2) [inline src/core/cita_bft.rs]: Chinese comma

**Review by u2 — CHANGES_REQUESTED** [review]

File: src/core/cita_bft.rs (hunk 2 above)
Comment (chaoticlonghair) [reply src/core/cita_bft.rs]: I don't think so.

Comment (u2) [reply src/core/cita_bft.rs]: `time-frame，`

File: src/core/cita_bft.rs (hunk 1 above)
Comment (chaoticlonghair) [reply src/core/cita_bft.rs:150]: No!
When there are no transactions, just insert an empty block without check.
And I already write comments for both scenes, you can find them in this commit, too.

Comment (u2) [reply src/core/cita_bft.rs:150]: Yes, I have found it after reviewed all. But I think that you should write more comments at here, as a newbie it's too hard to infer all the things.

**Comment by chaoticlonghair** [comment]
@kaikai1024
When the emphasis on `transaction`, I prefer to "transaction", because the "tx" is too short, it will confuse some readers; when stress on `body`, I think the abbr. is enough to signify.
It's more like a prefix.

I prefer to "the tx bodies" and "the bodies of transactions" than "the transaction bodies" and "the bodies of txs".
I think this is a reasonable presentation.

File: src/core/cita_bft.rs (hunk 2 above)
Comment (chaoticlonghair) [reply src/core/cita_bft.rs]: Oh, I see it, I will update this PR after all discussions are finished.

File: src/core/cita_bft.rs (hunk 1 above)
Comment (chaoticlonghair) [reply src/core/cita_bft.rs:150]: Writing one thing in two locations is redundant, and redundancies always bring ambiguity.
For high maintainability, comments should be always in the locations where the associated codes are.

For example, publishers send messages and subscribers subscribe message.
I was puzzled why there are lists of subscribers in the locations before definitions of publishers.
Do publishers CARE ABOUT who subscribe them?
So how can we decouple them?

I do NOT think more comments always are right things.

In other scenes like this but more complex, I think the best way is to provide different methods to do those.

**Review by kaikai1024 — APPROVED** [review]

**Review by u2 — APPROVED** [review]

**Pull request merged**